packages/ecdh/tests/ecdh.test.ts

```diff
@@ -0,0 +1,57 @@
+import { BarretenbergBackend } from '@noir-lang/backend_barretenberg';
+import { Noir } from '@noir-lang/noir_js';
+import { ProofData } from '@noir-lang/types';
+import { expect } from 'chai';
+import { randomBytes } from 'crypto';
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+import 'mocha';
+import circuit_ecdh from '../../../target/ecdh.json';
+
+function generatePrivateKey(): Uint8Array {
+  return randomBytes(32);
+}
+
+describe('ECDH Circuit Tests', function() {
+  let noir: Noir;
+  let backend: BarretenbergBackend;
+  let correctProof: ProofData;
+
+  beforeEach(async () => {
+    const circuitFile = readFileSync(resolve(__dirname, '../../../target/ecdh.json'), 'utf-8');
+    const circuit = JSON.parse(circuitFile);
+    backend = new BarretenbergBackend(circuit);
+    noir = new Noir(circuit, backend);
+  });
+
+  it('Should generate valid proof for correct input', async function() {
+    this.timeout(20000); // Optional: Increase timeout if needed
+
+    const pk1 = generatePrivateKey();
+    const pk2 = generatePrivateKey();
+
+    // Convert Uint8Array to regular arrays
+    const input = {
+      private_key1: Array.from(pk1),
+      private_key2: Array.from(pk2),
+    };
+
+    correctProof = await noir.generateProof(input);
+
+    // Assert that correctProof is defined and has the expected structure
+    expect(correctProof).to.not.be.undefined;
+    expect(correctProof.proof).to.be.instanceOf(Uint8Array);
+  });
+
+  it('Should verify valid proof for correct input', async function() {
+    this.timeout(20000); // Increase timeout if needed
+
+    expect(correctProof).to.not.be.undefined; // Ensure proof is generated
+    if (!correctProof) {
+      throw new Error('Proof was not generated in the previous test');
+    }
```

Comment:
Avoid making your unit tests depend on each others: it makes your test suite fragile and subject to the issue you are guarding against with `if (!correctProof)`.

I am not sure what is the default behavior of Mocha: running in parallel or serially.
If parallel, you will indeed run into issues if your second test runs before the first: which is why you check if the proof is defined.

Instead you could:
- re generate a proof in the verification test for you to verify (yes duplicate the code in previous test)
- or to avoid re generating the same proof and make your tests run longer, you can generate a proof in a `beforeAll` hook and use that in both your prove and verification tests

Reply:
hmm actually mocha seems to run serially by default.
https://mochajs.org/#available-root-hooks

> Available root hooks and their behavior:
  beforeAll: In serial mode (Mocha’s default), before all tests begin, once only
  
 So all this stuff is not needed
```suggestion
```

Then please explicit set `"parallel": false` in `.mocharc.json`